protocol/dhcp.go

```diff
@@ -74,9 +80,20 @@ const (
 //	FLAG_BROADCAST_MASK uint16 = (1 << FLAG_BROADCAST)
 )
 
+func getRandomXID() (uint32, error) {
```

Comment:
What's the advantage of using this way to generate a random uint32?

Reply:
It's a cryptographic secure random number generator. Not sure it really matters here (it's all about someone's ability to predict the xid), but it avoids linter warnings and it doesn't make the code too complicated.

Reply:
Got it. Thanks!